docs/tutorials/spec2006-tutorial-details.md

```diff
@@ -0,0 +1,636 @@
+
+### An Overview of Host System - gem5 Interactions
+![**Figure 1.**]( ../images/spec_tutorial_figure1.png "")
+A visual depict of how gem5 interacts with the host system.
+gem5 is configured to do the following: booting the Linux kernel, running the benchmark, and copying the SPEC outputs to the host system.
+However, since we are interested in getting the stats only for the benchmark, we will configure gem5 to exit after the kernel is booted, and then we reset the stats before running the benchmark.
+We use KVM for Linux booting process to quickly boot the system, and after the process is complete, we switch to the desired detailed CPU to run the benchmark.
+Similarly, after the benchmark is complete, gem5 exits to host, which allows us to get the stats at that point.
+After that, optionally, we switch the CPU back to KVM, which allows us to quickly write the SPEC output files to the host.
+
+**Important:** gem5 will output the stats again when the gem5 run is complete.
+Therefore, we will see two stats in one file in stats.txt.
+The stats of the benchmark is the the first part of stats.txt, while the second part of the file contains the stats of the benchmark AND the process of writing output files back to the host.
+We are only interested in the first part of stats.txt.
+
+
+## Documenting the Preparing Steps
+### Setting up the Experiment Folder
+We set up a folder to contain (almost) all materials of the experiment.
+We use git to keep track of changes in the folder.
+
+```sh
+mkdir spec2006-experiments
+cd spec2006-experiments
+git init
+```
+
+We need add a remote to the repository.
+
+```sh
+git remote add origin https://your-remote-add/spec-experiment.git
+```
+
+We document the root folder of the experiment in launch_spec2006_experiments.py as follows,
+
+```sh
+experiments_repo = Artifact.registerArtifact(
+    command = '',
+    typ = 'git repo',
+    name = 'experiment',
+    path =  './',
+    cwd = './',
+    documentation = 'local repo to run spec 2006 experiments with gem5 full system mode'
+)
+```
+
+We use .gitignore file to ignore changes of certain files or folders.
+In this experiment, we will use this .gitignore file,
+
+```
+*.pyc
+m5out
+.vscode
+results
+gem5art-env
+disk-image/packer
+disk-image/spec-2006/spec-2006-image/spec-2006
+disk-image/packer_cache
+disk-image/spec-2006/CPU2006v1.0.1.iso
+gem5
+linux-4.19.83/
+```
+
+Essentially, we will ignore files and folders that we use gem5art to keep track of them, or the presence of those files and folders do not affect the experiment's results.
+
+### Building gem5
+In this step, we download the source code and build gem5 v20.1.0.2.
+
+```sh
+git clone -b v20.1.0.2 https://gem5.googlesource.com/public/gem5
+cd gem5
+scons build/X86/gem5.opt -j8
+```
+
+We have two artifacts: one is the gem5 source code (the gem5 git repo), and the gem5 binary (`gem5.opt`).
+The documentation of this step would be how we get the source code and how we compile the gem5 binary.
```

Comment:
This line seems confusing and redundant given the above "In this step, we download..."

Reply:
removed